File: C++/C++Basic/09.While-Loop-Lab/08.graduation.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    string name;
    cin >> name;

    int grade = 0;
    int fail = 0;

    double sumGrade = 0;

    while (grade < 12)
    {
        double yearGrade;
        cin >> yearGrade;

        if (yearGrade < 4.0)
        {
            if (fail == 1)
            {
                break;
            }
            fail++;
        }

        sumGrade += yearGrade;
        grade++;
    }

    if (grade == 12)
    {
        cout.setf(ios::fixed);
        cout.precision(2);

        double average = sumGrade / 12;
        cout << name << " graduated. Average grade: " << average << endl;
    }
    else 
    {
         cout << name << " has been excluded at " << grade << " grade" << endl;
    }
}

File: Python/Basic/09.While-Loop-Lab/08.graduation.py
name = input()

completed_class = 1
failed = 0
sum_of_grade = 0


while True:
    grade = float(input())

    if grade < 4:
        failed += 1
        if failed > 1:
            print(f"{name} has been excluded at {completed_class} grade")
            break
        
        continue
    
    sum_of_grade += grade
    
    if completed_class == 12:
        average_grade = sum_of_grade / 12
        print(f"{name} graduated. Average grade: {average_grade:.2f}")
        break
    completed_class += 1

